### PR TITLE
Allow expressions in UniqueConstraint

### LINKED ISSUE #805: Expressions on UniqueConstraint aren't supported
# Bug report

## What's wrong

Django 4.0 added support for a [positional `expressions` argument](https://docs.djangoproject.com/en/4.0/ref/models/constraints/#expressions), but it doesn't look like it's supported here yet.

Example:

```
class User:
    ...
    class Meta:
        constraints = [
            models.UniqueConstraint(Upper("email"), name="email_key"),
        ]
```

Output:

```
mypy .
xxx/models.py:74: error: Too many positional arguments for "UniqueConstraint"
xxx/models.py:74: error: Argument 1 to "UniqueConstraint" has incompatible type "Upper"; expected "Sequence[str]"
xxx/migrations/0001_initial.py:111: error: Too many positional arguments for "UniqueConstraint"
xxx/migrations/0001_initial.py:112: error: Argument 1 to "UniqueConstraint" has incompatible type "Upper"; expected "Sequence[str]"
```

## How is that should be

Shouldn't output errors.

## System information

- OS: Mac OS
- `python` version: 3.10
- `django` version: 4.0
- `mypy` version: 0.930
- `django-stubs` version: 1.9.0
- `django-stubs-ext` version: 0.3.1

### PR DESCRIPTION
**Pull request by knyghty**
## Related issues

Closes #805

I'm not sure I did it right, but both using `expressions` and `fields` works with this at least, e.g.:

```python
class MyModel(models.Model):
    ...
    class Meta:
        constraints = [
            models.UniqueConstraint(Upper("foo"), name="foo_upper_key"),
            models.UniqueConstraint(fields=["bar"], name="bar_key"),
        ]
```

### CODE REVIEW

File: django-stubs/db/models/constraints.pyi [hunk 1]
```diff
@@ -27,9 +28,15 @@ class CheckConstraint(BaseConstraint):
     def __init__(self, *, check: Q, name: str) -> None: ...
 
 class UniqueConstraint(BaseConstraint):
+    expressions: Tuple[Combinable]
```
Comment (sobolevn) [inline django-stubs/db/models/constraints.pyi]: `Tuple[Combinable, ...]`?

Comment (knyghty) [reply django-stubs/db/models/constraints.pyi]: Yes, I think so, I wasn't aware of this syntax, thanks.

I assume this should also be changed for `fields`? I can do this here, but it's not completely relevant to this change.

File: django-stubs/db/models/constraints.pyi [hunk 2]
```diff
@@ -27,9 +28,15 @@ class CheckConstraint(BaseConstraint):
     def __init__(self, *, check: Q, name: str) -> None: ...
 
 class UniqueConstraint(BaseConstraint):
+    expressions: Tuple[Combinable, ...]
     fields: Tuple[str]
```
Comment (sobolevn) [inline django-stubs/db/models/constraints.pyi]: Yes, the same for fields 🙂

Comment (knyghty) [reply django-stubs/db/models/constraints.pyi]: 👍🏻

**Review by sobolevn — APPROVED** [review]
Thanks!

**Comment by linusg** [comment]
For anyone else finding this and wondering why it still doesn't work in 1.12.0: unfortunately this change was reverted/commented out in #909 :confused:

https://github.com/typeddjango/django-stubs/blob/c0414c7b025b77caf9f72dbaf9e8471e11aad085/django-stubs/db/models/constraints.pyi#L30-L48

**Pull request merged**